nifi-nar-bundles/nifi-iotdb-bundle/nifi-iotdb-processors/src/main/java/org/apache/nifi/processors/PutIoTDB.java

```diff
@@ -0,0 +1,286 @@
+                        .getProperty(RECORD_READER_FACTORY)
+                        .asControllerService(RecordReaderFactory.class);
+
+        String timeFieldProperty = processContext.getProperty(TIME_FIELD).getValue();
+        String schemaProperty = processContext.getProperty(SCHEMA).getValue();
+        String alignedProperty = processContext.getProperty(ALIGNED).getValue();
+        String maxRowNumberProperty = processContext.getProperty(MAX_ROW_NUMBER).getValue();
+
+        FlowFile flowFile = processSession.get();
+
+        String timeField = timeFieldProperty != null ? timeFieldProperty : "Time";
+        Boolean aligned = alignedProperty != null ? Boolean.valueOf(alignedProperty) : false;
+        int maxRowNumber = maxRowNumberProperty != null ? Integer.valueOf(maxRowNumberProperty) : 1024;
+
+        try (final InputStream inputStream = processSession.read(flowFile);
+             final RecordReader recordReader =
+                     recordParserFactory.createRecordReader(flowFile, inputStream, getLogger())) {
+            if (flowFile == null) {
+                inputStream.close();
+                recordReader.close();
+                processSession.transfer(flowFile, REL_SUCCESS);
+                return;
+            }
+
+            HashMap<String, Tablet> tablets;
+            boolean needInitFormatter;
+            Schema schema;
+            Tuple<Boolean, String> result;
+
+            result =
+                    schemaProperty != null
+                            ? validateSchemaAttribute(schemaProperty)
+                            : validateSchema(recordReader.getSchema());
+
+            if (!result.getKey()) {
+                getLogger().error(result.getValue());
+                inputStream.close();
+                recordReader.close();
+                processSession.transfer(flowFile, REL_FAILURE);
+                return;
+            } else {
+                if (result.getValue() != null) {
+                    getLogger().warn(result.getValue());
+                }
+            }
+
+            schema =
+                    schemaProperty != null
+                            ? new ObjectMapper().readValue(schemaProperty, Schema.class)
+                            : convertSchema(recordReader.getSchema());
+
+            List<String> fieldNames = schema.getFieldNames();
+
+            needInitFormatter = schema.getTimeType() != Schema.TimeType.LONG;
+
+            tablets = generateTablets(schema, maxRowNumber);
+            SimpleDateFormat timeFormatter = null;
+
+            Record record;
+
+            while ((record = recordReader.nextRecord()) != null) {
+                Object[] values = record.getValues();
+                if (timeFormatter == null && needInitFormatter) {
+                    timeFormatter = initFormatter((String) values[0]);
+                    if (timeFormatter == null) {
+                        getLogger().error("The format of time is not supported.");
```

Comment:
This message should include some additional details, such as the FlowFile and record number:
```suggestion
                        getLogger().error("{} Record [{}] time format not supported", flowFile, recordNumber);
```